experimental/sql/src/projection_optimizations.py

```diff
@@ -0,0 +1,191 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from xdsl.ir import Operation, MLContext, Region, Block
+from typing import List, Type, Optional
+from xdsl.dialects.builtin import ArrayAttr, StringAttr, ModuleOp, IntegerAttr
+
+from xdsl.pattern_rewriter import (RewritePattern, GreedyRewritePatternApplier,
+                                   PatternRewriteWalker, PatternRewriter,
+                                   op_type_rewrite_pattern)
+
+import dialects.rel_alg as RelAlg
+
+# This file contains several rewrites on the rel_alg level, all of which concern
+# projections. Currently, there are three rewriters. The first one introduces a
+# projection before any operator, the second one fuses subsequent projections,
+# and the third one removes projetions that are just identity mappings.
+
+
+@dataclass
+class ProjectionOptimizer(RewritePattern):
```

Comment:
I think that all of the member functions here could be static. Maybe the should rather be top-level functions (outside of any class)?

Reply:
I agree. I feel like there might be a step later on, where I add a `rel_alg_utils` file that contains these kinds of analysis functions.